src/devices/src/virtio/queue.rs

```diff
@@ -22,6 +25,29 @@ pub(super) const VIRTQ_DESC_F_WRITE: u16 = 0x2;
 // The Virtio Spec 1.0 defines the alignment of VirtIO descriptor is 16 bytes,
 // which fulfills the explicit constraint of GuestMemoryMmap::read_obj_from_addr().
 
+#[derive(Debug)]
+pub enum QueueError {
+    /// Descriptor index out of bounds.
+    DescIndexOutOfBounds(u16),
+    /// Attempted to write to an invalid address into the used ring.
+    UsedRing(GuestMemoryError),
```

Comment:
nit:
To avoid any confusion, I would update this comment with:

`
\\\ Attempted to do an invalid write into the used ring.
`

since we can also receive, for example, the `PartialBuffer` error, which doesn't exactly refer to an invalid address.

Reply:
Good catch. Updated the comment.